### PR TITLE
Enhancing the jsondb GetOptions so filter results based on a complex query filters against indexed properties.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by chirino** [comment]
I also want to make sure that when your using a comparison operator across different data types we honor the order defined at: https://firebase.google.com/docs/database/rest/retrieve-data#section-rest-ordered-data

File: app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java [hunk 1]
```diff
@@ -522,19 +529,19 @@ public void testSet() throws IOException {
...
+//        HashMap<String, Object> user = map(
+//            "data", new Object[]{"hi", 100, "other"}
+//        );
+//        jsondb.set("/", mapper.writeValueAsString(user));
+//        String result6 = jsondb.getAsString("", prettyPrint).trim();
+//        assertThat(result6).isEqualTo(load("result6.json").trim());
+//
+//        jsondb.set("/data/1", mapper.writeValueAsString("update"));
+//        String result7 = jsondb.getAsString("", prettyPrint).trim();
+//        assertThat(result7).isEqualTo(load("result7.json").trim());
```
Comment (zregvart) [inline app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java]: can this be removed?

File: app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java [hunk 2]
```diff
@@ -587,6 +594,108 @@ public void shouldAscertainPropertyPairExistence() {
...
+            "name", "u3",
+            "age", 21
+        )));
+
+        ////////////////////////////////////////
+        //
+        // Check filtering ops against strings..
+        //
+        ////////////////////////////////////////
+        assertThat(jsondb.getAsString("/users", new GetOptions().filter(
```
Comment (zregvart) [inline app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java]: might be a good idea to use [JSONAssert](http://jsonassert.skyscreamer.org/)

File: app/rest/jsondb/src/main/java/io/syndesis/jsondb/impl/JsonRecord.java [hunk 3]
```diff
@@ -22,13 +22,13 @@
 
     private final String path;
     private final String value;
-    private final int kind;
+    private final String ovalue;
```
Comment (zregvart) [inline app/rest/jsondb/src/main/java/io/syndesis/jsondb/impl/JsonRecord.java]: what do we gain by replacing int with String? seems to me like that would have negative performance

Comment (chirino) [reply app/rest/jsondb/src/main/java/io/syndesis/jsondb/impl/JsonRecord.java]: Type was removed.  odata is a new field used for something different

File: app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java (hunk 1 above)
Comment (chirino) [reply app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java]: I need to uncomment it.

File: app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java (hunk 2 above)
Comment (chirino) [reply app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java]: I would reduce test coverage.  We want to verify the order of things that jsondb returns.

**Pull request merged**